components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/util/EndpointUtil.java

```diff
@@ -1899,4 +1902,22 @@ private static Class<? extends OAuthAuthzRequest> getOAuthAuthzRequestClass() th
         }
         return oAuthAuthzRequestClass;
     }
+
+    /**
+     * Validate the response mode against the response type as per FAPI spec.
+     * Response mode required to be jwt if code response type is used.
+     * <a href="https://openid.net/specs/openid-financial-api-part-2-1_0.html#authorization-server">5.2.2-2.2</a>
+     *
+     * @param responseType response mode
+     * @param responseMode response type
+     * @throws OAuthProblemException when response mode is not valid
+     */
+    public static void validateFAPIAllowedResponseMode(String responseType, String responseMode)
+            throws OAuthProblemException {
+
+        if (!(CODE_IDTOKEN.equals(responseType) || (CODE.equals(responseType) && JWT.equals(responseMode)))) {
```

Comment:
isnt other types of JWT response modes accepted? (ex: query.jwt, fragment.jwt, form_post.jwt)

In the JARM spec it states `The default for response type "code" is "query.jwt"`. That means using `query.jwt` response mode in conjunction with `code` response type is similar to using code + jwt combination.

Reply:
In all the places the FAPI spec explicitly mentions that it requires response mode value to be jwt when code response type is used.

Reply:
Can we check this against the conformance suite and also reach out to the FAPI working group for guidance? I agree the spec mentions that, but it doesn't seem logical.

Reply:
The following test is in the conformance suite
```
FAPI1-Advanced-Final: ensure response_type code fails

"This test uses response_type=code in the authorization request, which (as a JARM response has not been requested)
 is not permitted in FAPI1-Advanced - only the hybrid flow ('response_type=code id_token') or JARM 
 ('response_type=code&response_mode=jwt') is allowed. The authorization server should show an error message that
  the response type is unsupported or the request is invalid (a screenshot of which should be uploaded) or the user 
  should be redirected back to the conformance suite with a correct error response, or an error could be returned from the 
  PAR endpoint.",
  ```